### ISSUE TITLE
[Issue]: HEVC transcoding flips video 180

### ISSUE THREAD

**Issue by wbste** [jellyfin/jellyfin#11102]
### Please describe your bug

Installed [installer/jellyfin_10.8.13_windows-x64.exe](https://repo.jellyfin.org/releases/server/windows/stable/installer/jellyfin_10.8.13_windows-x64.exe), running latest version of Roku app/firefox/chrome.

On some GoPro footage I have (HEVC) the footage gets flipped 180 during transcoding, which appears on both Firefox and the Roku app (both presumably because they don't have that codec). Checking the logs, every video gets this 180 rotation added at the last line. I noticed the same when my phone records with HEVC, so not a GoPro issue.

```
  Stream #0:0[0x1](eng): Video: hevc (Main) (hvc1 / 0x31637668), yuvj420p(pc, bt709), 3840x2160 [SAR 1:1 DAR 16:9], 59927 kb/s, 23.98 fps, 23.98 tbr, 24k tbn (default)
    Metadata:
      creation_time   : 2023-06-25T13:36:50.000000Z
      handler_name    : GoPro H.265
      vendor_id       : [0][0][0][0]
      encoder         : GoPro H.265 encoder
      timecode        : 13:36:01:11
    Side data:
      displaymatrix: rotation of -180.00 degrees
```

Is there a configuration I can tweak somewhere so it won't rotate the file during transcoding? Using Intel QSV. I'm trying to avoid encoding the file to something else.

### Reproduction Steps

1. Setup Jellyfin with Intel QSV.
2. Load up HEVC from GoPro 10.
3. Stream to roku app or firefox.
4. Stream to Chrome or Edge to see it without transcoding.

### Jellyfin Version

10.8.13

### Environment

```markdown
- OS: Windows 11 Pro
- Linux Kernel: N/AA
- Virtualization: N/A
- Clients: Roku
- Browser: Firefox
- FFmpeg Version: Built into "installer" app (5.1.4-Jellyfin in C:\Program Files\Jellyfin\Server)
- Playback Method: Transcode
- Hardware Acceleration: Intel QSV
- GPU Model: Intel UHD Graphics (Alder Lake-N 041 GT1)
- Plugins:
- Reverse Proxy:
- Base URL:
- Networking:
- Storage:
```

### Jellyfin logs

```shell
N/A
```

### FFmpeg logs

```shell
See above.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by felix920506** [comment]
Hi, please provide complete ffmpeg logs

**Comment by nyanmisaka** [comment]
Software `autorotate` is incompatible with hardware transcoding. This will require inserting a hardware `transpose` filter to be fixed in the future. Some HWA methods also require writing a hardware transpose filter from scratch.

**Comment by nyanmisaka** [comment]
It's not that simple. What this line means is that the memory layout of the video contains the rotated data, so stripping it out of the metadata is useless.

**Comment by nyanmisaka** [comment]
Not all local video players support utilizing this rotation metadata. However, transcoding is done to ensure maximum compatibility, so we choose to discard this metadata and not pass it to the player.

I took a quick look and found that we currently lack the `transpose_cuda` filter. Once implemented, we can enable the transpose function for all hardware transcoding in Jellyfin.

**Comment by gnattu** [comment]
> Hi @nyanmisaka , do you have a feel of where this fits in the priorities?

Everything required for this in ffmpeg is already landed in the latest release: https://github.com/jellyfin/jellyfin-ffmpeg/releases/tag/v6.0.1-4

The next part is that we need to enable them in the server side to fix this, but I'm not sure when it will finally landed yet.

**Comment by nyanmisaka** [comment]
We will be working on this after the 10.9 branch is cut. So this *should* be available in 10.10.
